packages/docs/src/components/EmbeddedExample.tsx

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable react/no-danger */
+import React, { useState } from 'react';
+import ReactDOMServer from 'react-dom/server';
+import Prism from 'prismjs';
+import { ArrowIcon } from '@cmsgov/design-system';
+
+interface EmbeddedExampleProps {
+  children: React.ReactElement;
+}
+
+/**
+ * Shows a code example as its rendered form as well as the HTML that is rendered for it
+ *
+ * Using ReactDOMServer's renderToString function to extract the HTML from a React child
+ * @see https://reactjs.org/docs/react-dom-server.html#rendertostring
+ *
+ * @todo HTML formatting is not great -- look into making it better
+ */
+const EmbeddedExample = ({ children }: EmbeddedExampleProps) => {
+  const [open] = useState(false);
+  const html = ReactDOMServer.renderToString(children);
+  const highlightedContent = Prism.highlight(html, Prism.languages.html);
+
+  return (
+    <section className="c-embedded-example">
+      <div className="ds-u-border--1 ds-u-padding--2">{children}</div>
+      <details open={open} className="c-embedded-example__details">
+        <summary className="ds-u-margin-y--1 ds-c-button ds-c-button--small ds-c-button--transparent ds-u-padding--1 ds-u-text-decoration--none">
+          <ArrowIcon direction="right" /> Code snippet
+        </summary>
+        <pre className="ds-u-margin-bottom--4 ds-u-overflow--auto ds-u-padding--2 language-html">
+          <code
+            dangerouslySetInnerHTML={{ __html: highlightedContent }}
+            className="language-html"
+          />
+        </pre>
```

Comment:
I know we talked about not generating the HTML from a React component but instead just having a code block in the content, but upon implementation, I didn't feel great about that. 1) Because we're generating lorem ipsum text, you'd have to copy the rendered markup from the browser 2) _if_ these examples ever change, both the example & and HTML code block would need to be updated.

So I attempted this implementation. Current biggest con is that the formatting of the code snippet is kind of rough and I haven't put a lot of effort into finding a way to fix that.

I also think this implementation may help with other, more complicated pages like [Flexbox](https://design.cms.gov/utilities/flexbox/) which has multiple, long examples. Being able to hide the code snippet helps save page space and may also help us break up these examples into smaller chunks / code snippets when we get there.

I am open to other thoughts!

Reply:
This is actually much simpler than I expected. I really like this component! You've set it up nicely, and MDX seems to make it much easier.

Reply:
I bet we could [use prettier](https://prettier.io/docs/en/api.html#prettierformatsource-options)

Reply:
I think static examples are a good idea for formatting and the ability to use inline comments.

Reply:
Now with prettier!
<img width="1291" alt="Screen Shot 2022-05-03 at 1 19 39 PM" src="https://user-images.githubusercontent.com/33579665/166536595-40630636-28f0-48f6-a98a-9ee3c1592f25.png">

Reply:
👨‍🍳👌